modules/dnn/src/darknet/darknet_io.cpp

```diff
@@ -568,23 +617,46 @@ namespace cv {
                             padding = kernel_size / 2;
 
                         CV_Assert(kernel_size > 0 && filters > 0);
-                        CV_Assert(current_channels > 0);
+                        CV_Assert(tensor_shape[0] > 0);
 
-                        setParams.setConvolution(kernel_size, padding, stride, filters, current_channels,
+                        setParams.setConvolution(kernel_size, padding, stride, filters, tensor_shape[0],
                             batch_normalize);
 
-                        current_channels = filters;
+                        tensor_shape[0] = filters;
+                        tensor_shape[1] = (tensor_shape[1] - kernel_size + 2 * padding) / stride + 1;
+                        tensor_shape[2] = (tensor_shape[2] - kernel_size + 2 * padding) / stride + 1;
+                    }
+                    else if (layer_type == "connected")
+                    {
+                        int output = getParam<int>(layer_params, "output", 1);
+                        bool batch_normalize = getParam<int>(layer_params, "batch_normalize", 0) == 1;
+
+                        CV_Assert(output > 0);
+
+                        setParams.setFullyConnected(output, batch_normalize);
+
+                        if(layers_counter && tensor_shape[1] > 1)
+                            net->out_channels_vec[layers_counter-1] = total(tensor_shape);
+
+                        tensor_shape[0] = output;
+                        tensor_shape[1] = 1;
+                        tensor_shape[2] = 1;
```

Comment:
Looks like we need to resize `tensor_shape` to a single element here.

Reply:
That can be done, but I wanted to act it more like size of ** 1 * 1 * output ** instead of just **output**,.
Considering the case, if any other layer comes connected lets say `upsample` then width and height can simply be resized. If I resize it to single element, then for later layer we will have to check tensor_shape shape and then push_back.
I am not sure if this thing kinda model is there or not, but thought there was no harm doing that. 
I can change it single element size if you think that it will be better that way.